Add a new section "explaining" code and update changelog management. (#676)
returntocorp/semgrep-docs: docs/contributing/contributing-code.md

you would expect. Pull requests require approval of at least one maintainer and
[CI to be passing](https://github.com/returntocorp/semgrep/actions).

### Explaining code

It's important for code to be easy to maintain. This allows all of us to
spend more time on new features rather than spending it on studying
legacy code. As a general rule of thumb, assume that all context that
is not written down will be lost and forgotten. Useful context includes:

* Why does this code exist?
* What or who uses this code?
* What does this code achieve?
* Could this code be replaced by an off-the-shelf component? Why not?
* Does it implement a formal specification or a well-known pattern? Where can
  we learn more about it?

We ask that **each source file start with one comment** that
concisely answers these questions.

Here's a short example:
```
(*
   Generate unique names with a given prefix.
*)
```

> It could be improved by explaining a bit about the code's uses:

Reviewer: ```suggestion
It can be improved by explaining the code's uses:
```
